syntax = "proto3";

import "util.proto";

package collective_perception_cpp.proto;

/*
A SimulationSet stores sets of data from simulations (i.e., `SimPacket`s) with the same:
 - number of experiments,
 - number of agents,
 - number of steps,
 - communication range,
 - robot speed, and
 - swarm density
and with varying:
 - target fill ratios, and
 - sensor probabilities.
 */
message SimulationSet
{
    /*
    Currently the static and dynamic simulation has similar concepts but uses different variable names.
    For now they remain separated, but will be made uniform in the future.
    */

    // Common variables
    string sim_type = 1;
    uint32 num_agents = 2;
    uint32 num_trials = 3;
    repeated double tfr_range = 4 [packed = true]; // note that the Python version calls it `dfr_range`; will be updated in the future
    repeated double sp_range = 5 [packed = true];

    // Static (Python) simulation variables
    string comms_graph = 7;
    uint32 num_obs = 8; // will be merged to num_steps in the future
    uint32 comms_period = 9;
    float comms_prob = 10;

    // Dynamic (C++) simulation variables
    uint32 num_steps = 11;
    float comms_range = 12;
    float density = 13;
    float speed = 14;
    float assumed_sp = 15;
}

message SimulationStatsSet
{
    SimulationSet sim_set = 1;
    repeated StatsPacket stats_packets = 2;
}

message SimulationAgentDataSet
{
    SimulationSet sim_set = 1;
    repeated AgentDataPacket agent_data_packets = 2;
}